msg/templates/uorb_microcdr/microRTPS_client.cpp.em

```diff
@@ -86,54 +88,41 @@ void* send(void* /*unused*/)
     uint16_t header_length = 0;
 
     /* subscribe to topics */
-    int fds[@(len(send_topics))] = {};
-
-    // orb_set_interval statblish an update interval period in milliseconds.
 @[for idx, topic in enumerate(send_topics)]@
-    fds[@(idx)] = orb_subscribe(ORB_ID(@(topic)));
-    orb_set_interval(fds[@(idx)], _options.update_time_ms);
+    uORB::Subscription @(topic)_sub{ORB_ID(@(topic))};
 @[end for]@
 
     // ucdrBuffer to serialize using the user defined buffer
     ucdrBuffer writer;
-    header_length=transport_node->get_header_length();
-    ucdr_init_buffer(&writer, (uint8_t*)&data_buffer[header_length], BUFFER_SIZE - header_length);
+    header_length = transport_node->get_header_length();
+    ucdr_init_buffer(&writer, reinterpret_cast<uint8_t*>(&data_buffer[header_length]), BUFFER_SIZE - header_length);
```

Comment:
Why is this a reinterpret_cast?

Reply:
You cannot `static_cast` a `char *` pointer to a `unsigned char*` pointer, as `static_cast` does not support casting between different pointer types:
```console
src/modules/micrortps_bridge/micrortps_client/micrortps_client/microRTPS_client.cpp:144:80: error: invalid static_cast from type ‘char*’ to type ‘uint8_t* {aka unsigned char*}’
     ucdr_init_buffer(&writer, static_cast<uint8_t*>(&data_buffer[header_length]), BUFFER_SIZE - header_length);
                                                                                ^
compilation terminated due to -Wfatal-errors.
```
Hence, you use C type casting, or in a more elegant way, you use C++ `reinterpret_cast`, since you know exactly what type to expect while reading the address of that pointer.